### DISCUSSION TITLE
Support request

### DISCUSSION THREAD

**Discussion by tonielli** [semuconsulting/PyGPSClient#78] [General]
Hi all, I premise that I am a neophyte. I have tried installing pygpsclient on Raspberry py 4 on which a SparkFun GPS-RTK Dead Reckoning pHAT for Raspberry Pi is mounted, this board connects on Raspberry's ttyS0 at 38400. I have tried with Rasbian 32bit based Debian Bullseye and on Rasbian 64bit based Debian Bullseye, in the first case the installation fails because the system requires RUST, but even after installing the latter, nothing changes. In the second case the installation is successful but I can't get the privileges to read and write the ttyS0 port. Do you know how I can solve this problem?
Can you tell me the procedure?

Thank you 
Renato

**Comment by semuadmin** [comment]
Hi @tonielli 

Neophytes very welcome, but this is probably more an RPi/Linux question than a PyGPSClient question.

I strongly recommend using the latest stock Raspberry Pi OS - 64-bit if you have the memory for it - but PyGPSClient should work fine on any ARM Linux distribution with Python >= 3.8 and tkinter >=8 6.

The installation instructions in the README.md expain any dependencies. There is no dependency on "RUST", so I'm not sure where that came from?

Serial port access is a common problem on Linux - it's an O/S issue rather than a PyGPSClient issue.

1. Firstly, make sure your user is a member of the tty and/or dialout groups e.g. `usermod -a -G tty myuser` (you'll have to log out and in for this to take effect). This normally resolves the problem.
2. If you still have issues, there is a quick temporary fix you can try: `sudo chmod 666 /dev/ttys0`.
3. Make sure that no other process is using the serial port (e.g. gpsd).
4. Failing that, try Googling 'accessing serial port on linux' for more tips - or get back to me here.

**Reply by tonielli** [reply]
_Neophytes very welcome, but this is probably more an RPi/Linux question than a PyGPSClient question._

Yes I understand that, but there is some confusion on the forums and I was hoping you would help me, which you did.

_There is no dependency on "RUST", so I'm not sure where that came from?_

I don't know why but the 32Bit version of Rusbian asked me to install RUST, which is why I then chose the 32bit version.

For the serial I found at this link https://alselectro.wordpress.com/2019/12/30/raspberry-pi-serial-port-configuration-solution-to-permission-denied/

I used these commands to disable the consol:
sudo systemctl stop serial-getty@ttyS0.service
sudo systemctl disable serial-getty@ttyS0.service
and
usermod -a -G tty myuser

and I run PyGPSControl with the --userport option and everything works.

I am now leaving for a survey when I get back I will apply on the ntrip to make a base and a rover.

Thanks
and it's a beautiful job

Renato

**Comment by semuadmin** [comment]
@tonielli

UPDATE!!!

FYI It appears there _is_ a dependency on a rust compiler on 32-bit platforms in certain circumstances - see:

https://github.com/semuconsulting/PyGPSClient/discussions/83#discussion-5473072

https://github.com/semuconsulting/PyGPSClient/issues/84#issuecomment-1666443832
